wordpress-mobile/WordPress-iOS
# Remote Config: Add ability to override remote config parameters

## Description

This PR refactors the Remote Config architecture and adds the ability to override them from the debug menu.

## Screenshots

|![Simulator Screen Shot - iPhone 14 Pro - 2023-03-27 at 08 53 54](https://user-images.githubusercontent.com/25306722/227864556-c5b0aebd-f838-4295-874f-4563ba64205a.png)|![Simulator Screen Shot - iPhone 14 Pro - 2023-03-27 at 08 54 45](https://user-images.githubusercontent.com/25306722/227864573-fb9c7e8f-870c-46c4-ac45-ec67c247d7cb.png)|
|-|-|

## Testing Instructions

1. Run the app
2. Go to the debug menu
3. Tap "Remote Config"
4. Make sure the displayed values match the `/mobile/remote-config` endpoint
5. Tap on any parameter
6. Type in a new value
7. Tap Back
8. Make sure a checkmark is displayed
9. Make sure the overridden value is reflected in the app
10. Go back to the Remote Config debug menu
11. Tap on the same parameter
12. Clear the user-defined value
13. Tap Back
14. Make sure a checkmark is no longer displayed
15. Make sure the original value is not reflected in the app

## Regression Notes
1. Potential unintended areas of impact
Places where remote config is used, like jetpack overlays and the blaze flow.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually

3. What automated tests I added (or what prevented me from doing so)
`RemoteParameterTests`

PR submission checklist:
